plaso/storage/redis/redis_store.py

```diff
@@ -0,0 +1,489 @@
+# -*- coding: utf-8 -*-
+"""Redis store.
+
+Only supports task storage at the moment.
+"""
+from __future__ import unicode_literals
+
+import uuid
+
+import redis
+
+from plaso.lib import definitions
+from plaso.storage import identifiers
+from plaso.storage import interface
+from plaso.storage import logger
+
+
+class RedisStore(interface.BaseStore):
+  """Redis store.
+
+  Attribute containers are stored as Redis Hashes.
+  All keys are prefixed with the session identifier to avoid collisions.
+  Event identifiers are also stored in an index to enable sorting.
+  """
+
+  _FORMAT_VERSION = '20181013'
+  _EVENT_INDEX_NAME = 'sorted_event_identifier'
+  _FINALIZED_KEY_NAME = 'finalized'
+  _FINALIZED_BYTES = b'finalized'
+  _MERGING_KEY_NAME = 'merging'
+  _MERGING_BYTES = b'merging'
+  # DEFAULT_REDIS_URL is public so that it appears in generated documentation.
+  DEFAULT_REDIS_URL = 'redis://127.0.0.1/0'
+
+  def __init__(
+      self, storage_type=definitions.STORAGE_TYPE_TASK,
+      session_identifier=None, task_identifier=None):
+    """Initializes a Redis store.
+
+    Args:
+      storage_type (Optional[str]): storage type.
+      session_identifier (Optional[str]): session identifier, formatted as
+          a UUID.
+      task_identifier (Optional[str]): unique identifier of the task the store
+          will store containers for. If not specified, an identifier will be
+          generated.
+
+    Raises:
+      ValueError: if the storage type is not supported.
+    """
+    if storage_type != definitions.STORAGE_TYPE_TASK:
+      raise ValueError('Unsupported storage type: {0:s}.'.format(
+          storage_type))
+    super(RedisStore, self).__init__()
+    if not session_identifier:
+      session_identifier = str(uuid.uuid4())
+    self._session_identifier = session_identifier
+    if not task_identifier:
+      task_identifier = str(uuid.uuid4())
+    self._task_identifier = task_identifier
+    self._redis_client = None
+    self.serialization_format = definitions.SERIALIZER_FORMAT_JSON
+
+  @classmethod
+  def _SetClientName(cls, redis_client, name):
+    """Attempts to sets a Redis client name.
+
+    This method ignores errors from the Redis server or exceptions
+    indicating the method is missing, as setting the name is not a critical
+    function, and it is not currently supported by the fakeredis test library.
+
+    Args:
+      redis_client (Redis): an open Redis client.
+      name (str): name to set.
+    """
+    try:
+      redis_client.client_setname(name)
+    except redis.ResponseError as exception:
+      logger.debug(
+          'Unable to set redis client name: {0:s} with error: {1!s}'.format(
+              name, exception))
+
+  def _AddAttributeContainer(
+      self, container_type, container, serialized_data=None):
+    """Adds an attribute container to the store.
+
+   Args:
+      container_type (Type): container type attribute of the container being
```

Comment:
is `Type` correct? Based on `_GenerateRedisKey` and later references to container_type I suspect this should be `str`.